package com.radiance.mixins.vulkan_render_integration;

import com.radiance.client.texture.AuxiliaryTextures;
import net.minecraft.client.renderer.texture.atlas.SpriteSource.Output;
import net.minecraft.client.renderer.texture.atlas.SpriteSource.SpriteSupplier;
import net.minecraft.client.renderer.texture.atlas.sources.Unstitcher;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.server.packs.resources.ResourceManager;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Redirect;

@Mixin(Unstitcher.class)
public class UnstitchAtlasSourceMixins {

    @Redirect(method = "run(Lnet/minecraft/server/packs/resources/ResourceManager;"
        + "Lnet/minecraft/client/renderer/texture/atlas/SpriteSource$Output;)V",
        at = @At(value = "INVOKE",
            target = "Lnet/minecraft/client/renderer/texture/atlas/SpriteSource$Output;add("
                + "Lnet/minecraft/resources/ResourceLocation;"
                + "Lnet/minecraft/client/renderer/texture/atlas/SpriteSource$SpriteSupplier;)V"))
    public void cancelPBRLoad(Output instance, ResourceLocation identifier,
        SpriteSupplier spriteRegion, ResourceManager resourceManager) {
        if (AuxiliaryTextures.shouldSkipAtlasSprite(resourceManager, identifier)) {
            return;
        }
        instance.add(identifier, spriteRegion);
    }
}
